zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java

```diff
@@ -79,7 +81,13 @@ public void updateEntity(final Record<Form> record, final FormEntity entity) {
 
   @Override
   public void flush(final FormEntity entity, final BatchRequest batchRequest) {
-    batchRequest.add(indexName, entity);
+    final Map<String, Object> updateFields = new HashMap<>();
+    String id = entity.getId();
+    if (entity.getIsDeleted() != null && entity.getIsDeleted()) {
+      updateFields.put("isDeleted", true);
+      id = String.valueOf(entity.getKey());
```

Comment:
❓ Why we have here a different ID then?

Reply:
The change in ID is for the upsert to point to the previous Form record. Form records are not being updated, but overriden by providing a version increment.